### ISSUE TITLE
[Bug]: MO Crash by out of memory when running oltp update high concurrentcy test and tpcc test

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#8373]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 16acd36
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2023/03/11 02:02:24.985817 +0800 INFO frontend/util.go:414 time of Exec.Run : 7.091132ms connectionId 1095
2023/03/11 02:02:24.985916 +0800 INFO frontend/util.go:399 query trace status {"connection_id": 1095, "statement": "execute __mo_stmt_id_13 using @__mo_stmt_var_0,@__mo_stmt_var_1,@__mo_stmt_var_2", "status": "success", "span": {"trace_id": "b655b65f-bf6d-11ed-abc1-b07b25f8b524", "kind": "statement"}, "session_info": "connectionId 1095"}
2023/03/11 02:02:24.986199 +0800 INFO mpool/mpool.go:81 MPool global new high watermark
     allocations : 342482045
     frees : 208112057
     alloc bytes : 2051522644725
     free bytes : 952010962779
     current bytes : 1099511681946
     high water mark : 1099511748770

2023/03/11 02:02:24.986230 +0800 INFO frontend/util.go:399 query trace {"connection_id": 1095, "query": "execute __mo_stmt_id_14 using @__mo_stmt_var_0,@__mo_stmt_var_1,@__mo_stmt_var_2", "vars": "1 , 9 , 665", "session_info": "connectionId 1095"}
2023/03/11 02:02:24.986249 +0800 ERROR mpool/mpool.go:511 error: error: out of memory
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Alloc
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/common/mpool/mpool.go:511
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Realloc
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/common/mpool/mpool.go:592
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Grow
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/common/mpool/mpool.go:658
github.com/matrixorigin/matrixone/pkg/container/vector.extend
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/container/vector/tools.go:151
github.com/matrixorigin/matrixone/pkg/container/vector.appendOneFixed[...]
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/container/vector/vector.go:1210
github.com/matrixorigin/matrixone/pkg/container/vector.GetUnionOneFunction.func16
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/container/vector/vector.go:786
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*PartitionReader).Read
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/disttae/partition_reader.go:206
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*mergeReader).Read
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/disttae/reader.go:177
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).Run
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/pipeline/pipeline.go:70
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).Run
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/compile/scope.go:70
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func1
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/compile/scope.go:90
2023/03/11 02:02:24.986259 +0800 ERROR mpool/mpool.go:511 error: error: out of memory

**Comment by fengttt** [comment]
@aunjgr seems vector related.   please take a look.   thank you.

**Comment by aunjgr** [comment]
investigating

**Comment by LeftHandCold** [comment]
The case has been fixed in #8371

**Comment by aunjgr** [comment]
Please verify whether it's fixed by #8371

**Comment by aressu1985** [comment]
fixed
